### PR TITLE
Fix for linkable tabs ui breaking

### PR DESCRIPTION
**Pull request by hritesh04**
#18977 

fix for breaking changes 

### problem 

The issue was on macOS where the ui breaks when there is no parameters

### solution

in quiz file in line 59 the comparison should be with mac not Mac

This will fix the ui breaking if the issue was generated in macos

### CODE REVIEW

File: site/static/js/quiz.js [hunk 1]
```diff
@@ -1,57 +1,99 @@
...
+  });
+  // hide other answers
+  $(".quiz-instruction").addClass("hide");
+  // show the selected answer
+  $(".quiz-instruction[data-quiz-id='" + selectedId + "']").removeClass("hide");
+
+  const buttons = $(".option-row[data-quiz-id='" + selectedId + "']").find(
+    ".option-button"
+  );
+  //auto-select the first option for the user, to reduce the number of clicks
```
Comment (spowelljr) [inline site/static/js/quiz.js:26]:
```suggestion
  // auto-select the first option for the user, to reduce the number of clicks
```

File: site/static/js/quiz.js [hunk 2]
```diff
@@ -1,57 +1,99 @@
...
+    if (userOS === "mac") {
+      // use the name "macos" to match the button
+      userOS = "macos";
+    }
+    $(".option-row[data-level=0]").removeClass("hide");
+
+    const urlParams = new URLSearchParams(window.location.search);
+    const archParam = urlParams.get("arch");
+
+    //checks for query params and process each option one by one
```
Comment (spowelljr) [inline site/static/js/quiz.js:68]:
```suggestion
    // checks for query params and process each option one by one
```

**Review by spowelljr — CHANGES_REQUESTED** [review]
Just those two spaces and looks good to me

**Comment by hritesh04** [comment]
done

File: site/static/js/quiz.js [hunk 3]
```diff
@@ -1,57 +1,99 @@
...
+  $(".quiz-instruction").addClass("hide");
+  // show the selected answer
+  $(".quiz-instruction[data-quiz-id='" + selectedId + "']").removeClass("hide");
+
+  const buttons = $(".option-row[data-quiz-id='" + selectedId + "']").find(
+    ".option-button"
+  );
+  // auto-select the first option for the user, to reduce the number of clicks
+  if (buttons.length > 0) {
+    if (autoselect) {
```
Comment (spowelljr) [inline site/static/js/quiz.js:28]: Once last thing, this is unnecessarily double nested, could these two be `&&` together

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Comment by hritesh04** [comment]
done

**Review by spowelljr — APPROVED** [review]

**Pull request merged**